src/components/CytoscapeGraph/Layout/GroupCompoundLayout.ts

```diff
@@ -107,117 +110,119 @@ export default class GroupCompoundLayout {
    * This code gets executed on the cy.layout(...).run() is our entrypoint of this algorithm.
    */
   run() {
-    const { realLayout, compoundLayoutOptions } = this.options;
-    const parents = this.parents();
-    const children = parents.children();
-
-    // (1.a) Prepare parents by assigning a size and running the compound layout
-    parents.each(parent => {
-      const children = parent.children();
-      const targetElements = children.add(children.edgesTo(children));
-
-      // We expect a discrete layout here
-      const compoundLayout = targetElements.layout(compoundLayoutOptions);
-      compoundLayout.on('layoutstart layoutready layoutstop', _evt => {
-        // Avoid to propagate any local layout events up to cy, this would yield a global operation when not all nodes are ready.
-        return false;
-      });
-      compoundLayout.run();
+    [BoxByType.APP, BoxByType.NAMESPACE, BoxByType.CLUSTER].forEach(boxByType => {
+      const { realLayout, compoundLayoutOptions } = this.options;
+      const parents = this.parents(boxByType);
+      const children = parents.children();
 
-      // see https://github.com/cytoscape/cytoscape.js/issues/2402
-      const boundingBox = parent.boundingBox();
+      // (1.a) Prepare parents by assigning a size and running the compound layout
+      parents.each(parent => {
+        const children = parent.children();
+        const targetElements = children.add(children.edgesTo(children));
+
+        // We expect a discrete layout here
```

Comment:
The reason we expect a discrete layout is because we expect them to be finished after the "run" call. 
This could be extended to allow any type of non discrete layout that don't take too long to finish.

Reply:
I think it's fine as is, and we would want to use dagre I think.